From 7999ac0f4227c8737061cbd92bc9c2f5913ce0f0 Mon Sep 17 00:00:00 2001
From: Jason Montleon <jmontleo@redhat.com>
Date: Sun, 12 Oct 2025 17:38:20 -0400
Subject: [PATCH 0016/1240] Revert "clk: spacemit: ccu_pll: convert from
 round_rate() to determine_rate()"

This reverts commit d02c71cba7bba453d233a49497412ddbf2d44871.
---
 drivers/clk/spacemit/ccu_pll.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/drivers/clk/spacemit/ccu_pll.c b/drivers/clk/spacemit/ccu_pll.c
index d92f0dae65a4..45f540073a65 100644
--- a/drivers/clk/spacemit/ccu_pll.c
+++ b/drivers/clk/spacemit/ccu_pll.c
@@ -125,14 +125,12 @@ static unsigned long ccu_pll_recalc_rate(struct clk_hw *hw,
 	return entry ? entry->rate : 0;
 }
 
-static int ccu_pll_determine_rate(struct clk_hw *hw,
-				  struct clk_rate_request *req)
+static long ccu_pll_round_rate(struct clk_hw *hw, unsigned long rate,
+			       unsigned long *prate)
 {
 	struct ccu_pll *pll = hw_to_ccu_pll(hw);
 
-	req->rate = ccu_pll_lookup_best_rate(pll, req->rate)->rate;
-
-	return 0;
+	return ccu_pll_lookup_best_rate(pll, rate)->rate;
 }
 
 static int ccu_pll_init(struct clk_hw *hw)
@@ -154,6 +152,6 @@ const struct clk_ops spacemit_ccu_pll_ops = {
 	.disable	= ccu_pll_disable,
 	.set_rate	= ccu_pll_set_rate,
 	.recalc_rate	= ccu_pll_recalc_rate,
-	.determine_rate = ccu_pll_determine_rate,
+	.round_rate	= ccu_pll_round_rate,
 	.is_enabled	= ccu_pll_is_enabled,
 };
-- 
2.54.0

